hi-ml/src/health_ml/utils/reports.py

```diff
@@ -0,0 +1,165 @@
+
+    @staticmethod
+    def _remove_html_end(report_stream):
+        return report_stream.replace("</p>\n</div>\n</body>\n</html>", "")
+
+    def _create_template(self):
+        template_path = self.report_folder / "template.html"
+        template_path.touch(exist_ok=True)
+
+        self.template += """<!DOCTYPE html>
+<html lang="en">
+<head>
+<meta charset="UTF-8">
+<link href="https://cdn.jsdelivr.net/npm/bootstrap@5.1.3/dist/css/bootstrap.min.css" rel="stylesheet">
+<style type="text/css">
+    @page body{
+        size: A4 portrait;
+        @frame header_frame {
+            -pdf-frame-content: header_content;
+            left: 50pt; width: 512pt; top: 50pt; height: 40pt;
+        }
+        @frame content_frame {
+            left: 50pt; width: 512pt; top: 90pt; height: 632pt;
+        }
+        @frame footer_frame {
+            -pdf-frame-content: footer_content;
+            left: 50pt; width: 512pt; top: 772pt; height: 20pt;
+        }
+    }
+    tbody td {
+        text-align: center;
+        vertical-align: middle}
+    thead th {
+        text-align: center;}
+</style>
+<title> {{title}} </title>
+</head>
+<body>
+<div class="container-fluid justify-content-center align-items-center">
+<div class="container">
+<h1> {{title}} </h1>
+</div>
+<p>
+</p>
+</div>
+</body>
+</html>"""
+
+        return template_path
+
+    def add_table(self, df: pd.DataFrame):
+
+        self.template = self._remove_html_end(self.template)
+
+        num_existing_tables = self.template.count("table.to_html")
+        table_key = f"{TABLE_KEY_HTML}_{num_existing_tables}"  # starts at zero
+
+        self.template += """<div class="container" >
+{% for table in """ + table_key + """ %}
+    {{ table.to_html(classes=[ "table"], justify="center") | safe }}
+{% endfor %}
+</div>
+<br>
+</p>
+</div>
+</body>
+</html>"""
+
+        self.render_kwargs.update({table_key: [df]})
+
+    def add_image(self, image_path: str):
+        if image_path.startswith(str(self.report_folder)):
+            img_path_html = Path(image_path).relative_to(self.report_folder)
+        else:
+            img_path_html = Path(image_path)
+
+        image_path_pdf = Path(image_path)
+
+        self.template = self._remove_html_end(self.template)
+
+        image_key_html = IMAGE_KEY_HTML + "_0"
+        # Increment the image name so as not to replace other images
+        num_existing_images = self.template.count("<img src=")
+
+        image_key_html = image_key_html.split("_")[0] + f"_{num_existing_images}"
+
+        self.template += """<div class="container">
+{% for image_path in """ + image_key_html + """ %}
+    <img src={{image_path}} alt={{image_path}}>
+{% endfor %}
+</div>
+<br>
+</p>
+</div>
+</body>
+</html>"""
+
+        # Add these keys and paths to the keyword args for rendering later
+        self.render_kwargs.update({image_key_html: [img_path_html]})
+
+    def add_plot(self, plot_path: Optional[str] = None, fig: Optional[plt.Figure] = None):
```

Comment:
make plot_path helpful also when you provide a fig, so that I can control the file name?